src/hooks/useFocusTrap.ts

```diff
@@ -0,0 +1,46 @@
+// Based on https://hiddedevries.nl/en/blog/2017-01-29-using-javascript-to-trap-focus-in-an-element
+import { useRef, useEffect } from 'react';
+
+const KEYCODE_TAB = 9;
+
+export function useFocusTrap(visible = true) {
+    const elRef = useRef(null);
+    const handleFocus = (e: KeyboardEvent) => {
+        var focusableEls = elRef.current.querySelectorAll(
+                'a[href], button, textarea, input[type="text"], input[type="radio"], input[type="checkbox"], select'
```

Comment:
Need to consider any element with tab index of 0, input[type="range"], please look through Octuple and add any that might be missing.

Reply:
perhaps something like this, but typed :) --> https://zellwk.com/blog/keyboard-focusable-elements/